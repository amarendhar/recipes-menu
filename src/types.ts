export type Maybe<T> = T | null;

export type Recipe = {
  id: string;
  title: string;
  image: string;
  tags: string[];
  description: string;
  chefName: string;
};
