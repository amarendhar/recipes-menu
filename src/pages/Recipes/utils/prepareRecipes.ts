import { Entry, EntryFields, EntrySkeletonType } from "contentful";
import { Maybe, RecipesEntry, RecipesItemData } from "types";

type RecipeFields = {
  title: EntryFields.Text;
  photo: EntryFields.AssetLink;
  tags: EntryFields.Array<EntryFields.EntryLink<EntrySkeletonType>>;
  description: EntryFields.Text;
  chef: EntryFields.EntryLink<EntrySkeletonType>;
};

export const prepareRecipes = (
  data: Maybe<RecipesEntry>
): RecipesItemData[] => {
  if (!data?.items) {
    return [];
  }

  const recipes: RecipesItemData[] = data.items.map(
    (recipe: Entry<EntrySkeletonType, undefined, string>) => {
      const fields = recipe?.fields as RecipeFields;

      const id: string = recipe.sys.id;
      const title = fields?.title || "";
      const image = (fields?.photo?.fields?.file?.url || "") as string;
      const rating = 0;

      const recipeData: RecipesItemData = {
        id,
        title,
        image,
        rating,
      };

      return recipeData;
    }
  );

  return recipes;
};
