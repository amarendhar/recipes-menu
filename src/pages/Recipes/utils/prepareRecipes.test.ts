import { prepareRecipes } from "./prepareRecipes";
import { RecipesEntry } from "types";
import mockRecipes from "mocks/__fixtures__/mockRecipes.json";

describe("# prepareRepository", () => {
  it("Should return prepared recipes", () => {
    const recipes = prepareRecipes(mockRecipes as RecipesEntry);

    recipes.forEach((recipe) => {
      expect(recipe).toEqual({
        id: expect.any(String),
        title: expect.any(String),
        image: expect.any(String),
      });
    });

    expect(recipes).toMatchSnapshot();
  });
});
