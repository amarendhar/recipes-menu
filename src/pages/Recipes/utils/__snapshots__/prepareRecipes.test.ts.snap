// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# prepareRepository Should return prepared recipes 1`] = `
Array [
  Object {
    "id": "4dT8tcb6ukGSIg2YyuGEOm",
    "image": "//images.ctfassets.net/kk2bw5ojx476/61XHcqOBFYAYCGsKugoMYK/0009ec560684b37f7f7abadd66680179/SKU1240_hero-374f8cece3c71f5fcdc939039e00fb96.jpg",
    "rating": 0,
    "title": "White Cheddar Grilled Cheese with Cherry Preserves & Basil",
  },
  Object {
    "id": "5jy9hcMeEgQ4maKGqIOYW6",
    "image": "//images.ctfassets.net/kk2bw5ojx476/48S44TRZN626y4Wy4CuOmA/9c0a510bc3d18dda9318c6bf49fac327/SKU1498_Hero_154__2_-adb6124909b48c69f869afecb78b6808-2.jpg",
    "rating": 0,
    "title": "Tofu Saag Paneer with Buttery Toasted Pita",
  },
  Object {
    "id": "2E8bc3VcJmA8OgmQsageas",
    "image": "//images.ctfassets.net/kk2bw5ojx476/3TJp6aDAcMw6yMiE82Oy0K/2a4cde3c1c7e374166dcce1e900cb3c1/SKU1503_Hero_102__1_-6add52eb4eec83f785974ddeac3316b7.jpg",
    "rating": 0,
    "title": "Grilled Steak & Vegetables with Cilantro-Jalape√±o Dressing",
  },
  Object {
    "id": "437eO3ORCME46i02SeCW46",
    "image": "//images.ctfassets.net/kk2bw5ojx476/5mFyTozvSoyE0Mqseoos86/fb88f4302cfd184492e548cde11a2555/SKU1479_Hero_077-71d8a07ff8e79abcb0e6c0ebf0f3b69c.jpg",
    "rating": 0,
    "title": "Crispy Chicken and Rice	with Peas & Arugula Salad",
  },
]
`;
